src/System.Text.Primitives/System/Text/InvariantParser_byte.cs

```diff
@@ -76,6 +78,12 @@ unsafe public static bool TryParse(byte* utf8Text, int index, int length, out by
                 {
                     value = candidate;
                 }
+                else // overflow
```

Comment:
Is this a reliable check for overflow? i.e are you sure that (value * 10 + digit) % 255 cannot be greater than value, which would mean the check is incorrect for at least parsing byte values.